src/os/bluestore/bluestore_tool.cc

```diff
@@ -1175,6 +1182,17 @@ int main(int argc, char **argv)
       exit(EXIT_FAILURE);
     }
     cout << sharding << std::endl;
+  } else if (action == "trim") {
+    BlueStore bluestore(cct.get(), path);
+    int r = bluestore.cold_open();
+    if (r < 0) {
+      cerr << "error from cold_open: " << cpp_strerror(r) << std::endl;
+      exit(EXIT_FAILURE);
+    }
+    cout << "bluestore offline trim begin" << std::endl;
+    bluestore.trim_free_space();
```

Comment:
Perhaps it's worth to add an optional device selection to be able to trim both all disks and selected one.

Reply:
done